features/integration/abi.feature

```diff
@@ -44,20 +44,120 @@ Feature: ABI Interaction
+      | referenceTest(account,application,account,asset,account,asset,asset,application,application)uint8[9] | Uabo7LuH+JfxYes3JwpJ4Fz3Kzoz5LYtZhWxaN8pa3g=,AAAAAAAAAAo=,Uabo7LuH+JfxYes3JwpJ4Fz3Kzoz5LYtZhWxaN8pa3g=,AAAAAAAAABQ=,1Hq2PnhdWWwQ3IvRrz0gZ8EZmoH68yc2DzLpab3P8uA=,AAAAAAAAABQ=,AAAAAAAAABU=,AAAAAAAAAAo=,AAAAAAAAAAs= | AQECAQECAAAB |
+
+  Scenario: Method call optin and closeout execution
+    Given a new AtomicTransactionComposer
+    When I create the Method object from method signature "optIn(string)string"
+    And I create a new method arguments array.
+    # "Algorand Fan"
+    And I append the encoded arguments "AAxBbGdvcmFuZCBGYW4=" to the method arguments array.
+    And I add a method call with the transient account, the current application, suggested params, on complete "optin", current transaction signer, current method arguments.
+    And I build the transaction group with the composer. If there is an error it is "".
+    Then The composer should have a status of "BUILT".
+    And I gather signatures with the composer.
+    Then The composer should have a status of "SIGNED".
+    And I execute the current transaction group with the composer.
+    Then The composer should have a status of "COMMITTED".
+    # "hello Algorand Fan"
+    And The app should have returned "ABJoZWxsbyBBbGdvcmFuZCBGYW4=".
+
+    Given a new AtomicTransactionComposer
+    When I create the Method object from method signature "closeOut()string"
+    And I create a new method arguments array.
+    And I add a method call with the transient account, the current application, suggested params, on complete "closeout", current transaction signer, current method arguments.
+    And I build the transaction group with the composer. If there is an error it is "".
+    Then The composer should have a status of "BUILT".
+    And I gather signatures with the composer.
+    Then The composer should have a status of "SIGNED".
+    And I execute the current transaction group with the composer.
+    Then The composer should have a status of "COMMITTED".
+    # "goodbye Algorand Fan"
+    And The app should have returned "ABRnb29kYnllIEFsZ29yYW5kIEZhbg==".
+
+  Scenario: Method call delete execution
+    Given I create a new transient account and fund it with 100000000 microalgos.
+    And I make a transaction signer for the transient account.
+    And I build an application transaction with the transient account, the current application, suggested params, operation "create", approval-program "programs/abi_method_call.teal.tok", clear-program "programs/one.teal.tok", global-bytes 0, global-ints 0, local-bytes 1, local-ints 0, app-args "", foreign-apps "", foreign-assets "", app-accounts "", extra-pages 0
```

Comment:
Great job in general!

It looks to me that you are creating another application (just to delete), right? I might need some help on this since this is not easy to do in JAVA (it will overwrite the previous created application and app-id), need some thinking about that...

Reply:
oh never mind, it seems I handled this...